myapp/app/controllers/admin/users_controller.rb

```diff
@@ -36,7 +36,11 @@ def update
     end
 
     def destroy
-      @user.destroy
+      ActiveRecord::Base.transaction do
+        Task.where(user: @user).in_batches(of: 500) { |tasks| tasks.delete_all }
```

Comment:
Though the `dependent: :destroy` will destroy tasks when user is deleted, but it will trigger N+1. Better delete in batches.

Reply:
If callback execution is not needed, it seems to me that `dependent: :delete_all` would be fine 👀 
https://guides.rubyonrails.org/association_basics.html#dependent